Lesson 1

JustSenoko opened the pull request:
Постаралась общие методы определить в абстрактных классах Animal и Obstacle, чтобы не копировать одинаковые куски кода в классы-наследники.
Несколько смущает, что в класс Robot пришлось скопировать почти все из класса Animal.

JustSenoko committed:
прописала gitignore

JustSenoko committed:
Добавила классы и их взаимодействия

usharik commented on src/ru/geekbrains/lesson1/participants/animals/Animal.java:
```diff
@@ -0,0 +1,87 @@
+    public void getInfo(boolean showSkills) {
+        String skills = "";
+        if (showSkills) {
+            if (this.maxRunLength > 0) {
```
Тут лучше использовать StringBuilder

JustSenoko replied:
исправила в новых коммитах

usharik replied:
Ok

usharik commented on src/ru/geekbrains/lesson1/Application.java:
```diff
@@ -0,0 +1,32 @@
+public class Application {
+
+    public static void main(String[] args) {
+        Team team = new Team(
```
Пожалуйста, доработайте код, чтобы задание было выполнено полностью

JustSenoko replied:
Почему-то у меня в прошлый раз закоммитилась недописанная версия, хотя после коммита я код не правила. Пока не могу разобраться в чем причина, в следующий раз буду сравнивать код в IDEA и на сервере

usharik replied:
Стоит взять себе за правило проверять код перед коммитом и после создания PR

JustSenoko committed:
Переделала String на StringBuilder
почему-то в прошлый раз закоммитилась не последняя версия (пользовалась SmartGit, теперь пробую через IDEA)

JustSenoko committed:
почему-то в прошлый раз закоммитилась не последняя версия (пользовалась SmartGit, теперь пробую через IDEA)

usharik commented on src/ru/geekbrains/lesson1/participants/Robot.java:
```diff
@@ -0,0 +1,90 @@
+    }
+
+    @Override
+    public void run(double length) {
```
В данном случае нормально что пришлось копировать код. Тут довольно модельный пример который должен показать разницу между абстрактными классами и интерфейсами.
